app/kubemci/pkg/kubeutils/utils.go

```diff
@@ -104,102 +99,38 @@ var getClientset = func(kubeconfigPath, contextName string) (kubeclient.Interfac
-		fmt.Println("Unable to sync kubernetes informers")
-	}
-
-	fmt.Println("Synced", b.svcSynced(), b.podSynced())
-}
-
 // GetProbe returns a probe that's used for the given nodeport
-func (t *KubernetesInformer) GetProbe(port ingressbe.ServicePort) (*api_v1.Probe, error) {
-	sl := t.svcLister.List()
-
-	// Find the label and target port of the one service with the given nodePort
-	var service api_v1.Service
-	var svcPort api_v1.ServicePort
-	var found bool
-OuterLoop:
-	for _, as := range sl {
-		service = *as.(*api_v1.Service)
-		for _, sp := range service.Spec.Ports {
-			svcPort = sp
-			// only one Service can match this nodePort, try and look up
-			// the readiness probe of the pods behind it
-			if int32(port.Port) == sp.NodePort {
-				found = true
-				break OuterLoop
-			}
-		}
-	}
-
-	if !found {
-		return nil, fmt.Errorf("unable to find nodeport %v in any service", port)
+func GetProbe(client kubeclient.Interface, port ingressbe.ServicePort) (*api_v1.Probe, error) {
+	svc, err := client.CoreV1().Services(port.SvcName.Namespace).Get(port.SvcName.Name, meta_v1.GetOptions{})
+	if err != nil {
+		fmt.Printf("Unable to find service %v in namespace %v\n", port.SvcName.Name, port.SvcName.Namespace)
+		return nil, err
 	}
 
-	return t.getHTTPProbe(service, svcPort.TargetPort, port.Protocol)
-}
-
-// isSimpleHTTPProbe returns true if the given Probe is:
-// - an HTTPGet probe, as opposed to a tcp or exec probe
-// - has no special host or headers fields, except for possibly an HTTP Host header
-func isSimpleHTTPProbe(probe *api_v1.Probe) bool {
-	return (probe != nil && probe.Handler.HTTPGet != nil && probe.Handler.HTTPGet.Host == "" &&
-		(len(probe.Handler.HTTPGet.HTTPHeaders) == 0 ||
-			(len(probe.Handler.HTTPGet.HTTPHeaders) == 1 && probe.Handler.HTTPGet.HTTPHeaders[0].Name == "Host")))
-}
-
-// geHTTPProbe returns the http readiness probe from the first container
-// that matches targetPort, from the set of pods matching the given labels.
-func (t *KubernetesInformer) getHTTPProbe(svc api_v1.Service, targetPort intstr.IntOrString, protocol ingressut.AppProtocol) (*api_v1.Probe, error) {
-	l := svc.Spec.Selector
+	selector := labels.SelectorFromSet(svc.Spec.Selector)
 
-	// Lookup any container with a matching targetPort from the set of pods
-	// with a matching label selector.
-	pl, err := t.podLister.List(labels.SelectorFromSet(labels.Set(l)))
+	pl, err := client.CoreV1().Pods(port.SvcName.Namespace).List(meta_v1.ListOptions{
+		LabelSelector: selector.String(),
+	})
 	if err != nil {
+		fmt.Printf("Unable to find pods backing service %v in namespace %v\n", port.SvcName.Name, port.SvcName.Namespace)
 		return nil, err
 	}
 
-	// If multiple endpoints have different health checks, take the first
-	sort.Sort(ingressct.PodsByCreationTimestamp(pl))
+	for _, pod := range pl.Items {
+		logStr := fmt.Sprintf("Pod %v matching service selectors %v (targetport %+v)", pod.Name, selector.String(), port.SvcTargetPort)
 
-	for _, pod := range pl {
```

Comment:
Same: Note the place where this is copied from and add a TODO to refactor to share and not duplicate.
We should be able to refactor this code to extract the readiness probe, given the pods.

cc @bowei